test-suite-data.json

```diff
@@ -622,5 +622,77 @@
     "qualifiers": null,
     "subpath": null,
     "is_invalid": false
+  },
+  {
+    "description": "bazel module with default registry",
+    "purl": "pkg:bazel/rules_java@7.8.0?registry_url=https://bcr.bazel.build/",
+    "canonical_purl": "pkg:bazel/rules_java@7.8.0",
+    "type": "bazel",
+    "namespace": null,
+    "name": "rules_java",
+    "version": "7.8.0",
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "bazel module with custom registry",
+    "purl": "pkg:bazel/rules_java@7.8.0?registry_url=https://example.org/bazel-registry/",
+    "canonical_purl": "pkg:bazel/rules_java@7.8.0?registry_url=https://example.org/bazel-registry",
+    "type": "bazel",
+    "namespace": null,
+    "name": "rules_java",
+    "version": "7.8.0",
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "bazel module with relaxed semver version",
+    "purl": "pkg:bazel/curl@8.8.0.bcr.1",
+    "canonical_purl": "pkg:bazel/curl@8.8.0.bcr.1",
+    "type": "bazel",
+    "namespace": null,
+    "name": "curl",
+    "version": "8.8.0.bcr.1",
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "bazel module names are lowercased",
+    "purl": "pkg:bazel/Curl@8.8.0.bcr.1",
+    "canonical_purl": "pkg:bazel/Curl@8.8.0.bcr.1",
+    "type": "bazel",
+    "namespace": null,
+    "name": "curl",
+    "version": "8.8.0.bcr.1",
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": true
```

Comment:
I am  not sure we want this as invalid. I think that the canonical purl name should be lowercased as part canonicalization.

Reply:
Should I also keep casing out of canonicalization? `pkg:bazel/Curl@8.8.0.bcr.1` could be a valid canonical PURL, just not one for a module Bazel would ever accept.